### ISSUE TITLE
Bottlerocket support

### ISSUE THREAD

**Issue by valorl** [awslabs/node-latency-for-k8s#21]
This looks like a really nice tool, but unfortunately it doesn't work on Bottlerocket nodes, where `/var/log/messages` isn't present,. AFAICT, there are only systemd/journal logs, which would probably require an external dependency / bindings to read.
That being said, I've tried to find some of the events defined here in `journalctl` logs and was unsuccessful. So perhaps this is not possible to make work on Bottlerocket or requires a completely different approach?

**Comment by bwagner5** [comment]
I suspected it wouldn't work out of the box with Bottlerocket. But the nice thing is that the timing events it looks for are easily modifiable in the code and can gracefully fallback depending on which events it is able to find. I haven't exposed configuring the events without directly modifying the code, but it's possible that we could expose a CRD or config map that exposes configurable sources (log files, http apis, etc) and event finds (regex, api requests, etc). 

I should have some time soon to play with this on bottlerocket and at least setup the timing events in the code so that it will work. The configuration stuff will likely take longer and not worth it just to extend to bottlerocket. 

Having said that, PRs are certainly welcome if you wanted to take a crack at the code modifications. Or if you want to collaborate on this issue, it would be helpful to know what sources (log files) and timing regex's could be used to find the timing points.

**Comment by valorl** [comment]
Yeah, the thing is I wasn't able to find any equivalent events on Bottlerocket, though I only had a look at systemd logs. Not sure where else to even look. If you have any ideas, let me know.

**Comment by arnaldo2792** [comment]
Hey @valorl , you can filter out the kernel messages with `journalctl -k | grep "Linux version"`:

```bash
bash-5.1# journalctl -k | grep "Linux version"
Jan 30 17:25:11 localhost kernel: Linux version 5.10.157 (builder@buildkitsandbox) (x86_64-bottlerocket-linux-gnu-gcc (Buildroot 2022.11) 11.3.0, GNU ld (GNU Binutils) 2.38) #1 SMP Fri Jan 27 18:01:57 UTC 2023
```

**Comment by valorl** [comment]
@arnaldo2792 thanks, I've tried that (and other queries for non `-k` logs) but I think maybe I was on a node that's too old and logs might have been deleted already

@bwagner5  I'm not sure when I'll be able to find some time to look at this, but any pointers for reading systemd logs in Go ? Couldn't find anything native, only https://github.com/coreos/go-systemd/tree/main/sdjournal

**Comment by bwagner5** [comment]
@valorl I briefly looked into some systemd log parsing libraries. Seems a bit overkill which is why I ended up just reading the files.

**Comment by bwagner5** [comment]
Probably will need to do this anyways for AL2023 since logs default to the journal

**Comment by bwagner5** [comment]
I've done some work to use the journal rather than looking directly at `/var/log/messages`. I implemented it with sdjournal first, which requires CGO... after much pain, I think it's better to build journalctl into the image and exec it...  It's not quite ready yet, but wanted to give an update here.

**Comment by fahaddd-git** [comment]
thanks! Interested in using this for ubuntu

**Comment by youwalther65** [comment]
@bwagner5 Any progress here? AL2023 is coming soon hopefully as well! Thank you!
